project-ideas/order-ledger/createSalesOrder.md

```diff
@@ -1,47 +1,210 @@
 ### co.hotwax.orderledger.order.OrderServices.create#SalesOrder (Application Layer - OMS)
-This service will take in the order JSON in OMSNewOrdersFeed and set up a complete order by performing any surrounding crud operations as needed.
+
+This is the application layer service which will be responsible for
+1. handling multiple business rules like existence checks, validating customer, facility, products, etc.
+2. coordinating calls to multiple lower level services in the right sequence to achieve the order creation. This includes create Customer, Contact Mechs etc.
+
+This service will take in the order JSON transformed from the response received from an external API request (REST/GraphQL) and create the payload for the core create#org.apache.ofbiz.order.order.OrderHeader service by performing any surrounding validations and crud operations as needed.
+
 1. Parameters
     * Input Parameters
-        * orderJson (Map)
-2. Initialize orderContext
-3. Set following to orderContext
```

Comment:
removed these, since this can be taken care of in mapping service, the input map to create#SalesOrder should already have these values set